src/Deprecated.jl

```diff
@@ -52,3 +52,118 @@ const AbstractBigDataEntry = AbstractDataEntry
+InMemoryDataStore(args...; kwargs...) = error("`InMemoryDataStore` is deprecated, use TBD")#TODO
+
+_uniqueKey(dfg::AbstractDFG, v::AbstractDFGVariable, key::Symbol)::Symbol = error("_uniqueKey is deprecated")
+
+getDataBlob(store::AbstractDataStore, entry::AbstractDataEntry) = error("AbstractDataStore $(typeof(store)) is deprecated.")
+addDataBlob!(store::AbstractDataStore, entry::AbstractDataEntry, data) = error("AbstractDataStore $(typeof(store)) is deprecated.")
+updateDataBlob!(store::AbstractDataStore, entry::AbstractDataEntry, data)  = error("AbstractDataStore $(typeof(store)) is deprecated.")
+deleteDataBlob!(store::AbstractDataStore, entry::AbstractDataEntry) = error("AbstractDataStore $(typeof(store)) is deprecated.")
+listDataBlobs(store::AbstractDataStore) = error("AbstractDataStore $(typeof(store)) is deprecated.")
+
+
+
+
+"""
+$(SIGNATURES)
+Add Big Data Entry to distributed factor graph.
+Should be extended if DFG variable is not returned by reference.
+
+Example
+
+See docs for `getDataEntryElement`.
+
+Related
+
+addData!, getDataEntryElement, fetchData
+"""
+function addData!(dfg::AbstractDFG,
+                  lbl::Symbol,
+                  datastore::Union{FileDataStore, InMemoryDataStore},
+                  descr::Symbol,
+                  mimeType::AbstractString,
+                  data::Vector{UInt8} )
+  #
+  node = isVariable(dfg, lbl) ? getVariable(dfg, lbl) : getFactor(dfg, lbl)
+  # Make a big data entry in the graph - use JSON2 to just write this
+  entry = GeneralDataEntry(dfg, node, descr, mimeType=mimeType)
+  # Set it in the store
+  addDataBlob!(datastore, entry, data)
+  # Add the entry to the graph
+  addDataEntry!(node, entry)
+end
+# const addDataEntry! = addData!
+
+
+
+"""
+$SIGNATURES
+Get both the entry and raw data element from datastore returning as a tuple.
+
+Notes:
+- This is the counterpart to `addDataEntry!`.
+- Data is identified by the node in the DFG object `dfglabel::Symbol` as well as `datalabel::Symbol`.
+- The data should have been stored along with a `entry.mimeType::String` which describes the format of the data.
+- ImageMagick.jl is useful for storing images in png or jpg compressed format.
+
+Example
+
+```julia
+# some dfg object
+fg = initfg()
+addVariable!(fg, :x0, IIF.ContinuousScalar) # using IncrementalInference
+# FileDataStore (as example)
+datastore = FileDataStore(joinLogPath(fg,"datastore"))
+
+# now some data comes in
+mydata = Dict(:soundBite => randn(Float32, 10000), :meta => "something about lazy foxes and fences.")
+
+# store the data
+addDataEntry!( fg, :x0, datastore, :SOUND_DATA, "application/json", Vector{UInt8}(JSON2.write( mydata ))  )
+
+# get/fetch the data
+entry, rawData = fetchData(fg, :x0, datastore, :SOUND_DATA)
+
+# unpack data to original format (this must be done by the user)
+@show entry.mimeType # "applicatio/json"
+userFormat = JSON2.read(IOBuffer(rawData))
+```
+
+Related
+
+addDataEntry!, addData!, fetchData, fetchDataEntryElement
+"""
```

Comment:
if we deprecating this, what is the direct replacement for this documented example in the new API -- specifically the combo?
```julia
addDataEntry!( fg, :x0, datastore, :SOUND_DATA, "application/json", Vector{UInt8}(JSON2.write( mydata ))  )
entry, rawData = fetchData(fg, :x0, datastore, :SOUND_DATA)
```

Reply:
I don't know what default to use yet. For persistence: the `FolderStore`, for quick and easy (not persisted or copied to the cloud) I like the in-memory data entry.

We currently have:
```julia
dataset1 = rand(UInt8, 1000)

ade,adb = addData!(InMemoryDataEntry, dfg, :x1, :random, dataset1)
#or
ade,adb = addData!(FileDataEntry, dfg, :x1, :random, "/tmp/dfgFilestore", dataset1)
#or
ds = FolderStore{Vector{UInt8}}(:filestore, "/tmp/dfgFilestore")
addBlobStore!(dfg, ds)
ade,adb = addData!(dfg, :filestore, :x1, :random, dataset1)
```
The docs are a bit behind and we need good examples. The main purpose of PR was to have a clean API.

Reply:
Direct replacement should be (if the store is not saved in the DFG):
```julia
blobstore = FolderStore{Vector{UInt8}}(:filestore, "/tmp/dfgFilestore")
addData!(fg, blobstore, :x0, :SOUND_DATA, Vector{UInt8}(JSON2.write( mydata )), mimeType="application/json")

rawData, entry = getData(fg, blobstore, :x0, :SOUND_DATA)
```

Reply:
Sorry for changing the parameter order. The DFG and BlobStore are mutated so they should go first and there were ambiguities with dispatch.

The idea is to only need the CRUD verbs `getData/addData!/updateData!/deleteData!` and not have other verbs such as fetch.

Reply:
CRUD API
- Verbs: get,add,update,delete
- Nouns: Data, DataBlob, DataEntry

Reply:
Ah, gotcha thanks!  This helps

Reply:
> Sorry for changing the parameter order.

Totally fine, I'm just trying to follow